feathr-impl/src/main/scala/com/linkedin/feathr/offline/source/pathutil/TimeBasedHdfsPathAnalyzer.scala

```diff
@@ -70,7 +70,18 @@ private[offline] class TimeBasedHdfsPathAnalyzer(pathChecker: PathChecker, dataL
   def analyze(filePath: String, timePartitionPattern: String): PathInfo = {
     val basePath = if (filePath.endsWith("/") || filePath.endsWith("=")) filePath else filePath + "/"
     val dateTimeResolution = OfflineDateTimeUtils.getDateTimeResolutionFromPattern(timePartitionPattern)
-    PathInfo(basePath, dateTimeResolution, timePartitionPattern)
+    if (basePath.endsWith(dailyFolder)) {
+      PathInfo(basePath, dateTimeResolution, timePartitionPattern)
+    } else if (basePath.endsWith(hourlyFolder)) {
+      PathInfo(basePath, dateTimeResolution, timePartitionPattern)
+    } else if (pathChecker.exists(basePath + dailyFolder)) {
+      PathInfo(basePath + dailyFolder, dateTimeResolution, timePartitionPattern)
+    } else if (pathChecker.exists(basePath + hourlyFolder)) {
+      PathInfo(basePath + hourlyFolder, dateTimeResolution, timePartitionPattern)
+    } else {
+      // Daily data can be Orc/Hive data following in HomeDir/datepartition=yyyy-MM-dd-00
+      PathInfo(basePath + "datepartition=", dateTimeResolution, "yyyy-MM-dd-00")
+    }
   }
 }
```

Comment:
Should we also concern below cases (not sure if they exist in real situations):
1. The base path doesn't end with daily/ or hourly/, the whole path is like: {somepath}/samples/2023/01/01/
2. The base path ends with daily/ but the time partition pattern is not /yyyy/MM/dd, the whole path is like: {somepath}/daily/2023-01-01

Reply:
One of the gradle test case failed at 'TestTimeBasedHdfsPathAnalyzer.scala:78' (you can find more details from below link). Could you solve that before merging?

Reply:
1. These are mainly meant for hdfs paths. I dont think paths of that pattern exist, and if they do we may need to write another pathAnalyzer.
2. Fixed the test, thanks for pointing it out.